src/jit/lclvars.cpp

```diff
@@ -2676,6 +2676,173 @@ int __cdecl Compiler::WtdRefCntCmp(const void* op1, const void* op2)
     return 0;
 }
 
+#if defined(_TARGET_AMD64_) && defined(FEATURE_CORECLR)
+
+/* static */
+int __cdecl Compiler::WtdRefCntCmpSimpleGC(const void* op1, const void* op2)
+{
+    LclVarDsc* dsc1 = *(LclVarDsc**)op1;
+    LclVarDsc* dsc2 = *(LclVarDsc**)op2;
+
+    if (!dsc1->lvTracked || !dsc2->lvTracked)
+    {
+        /* One or both untracked; sort tracked to earlier spot */
+        if (dsc1->lvTracked)
+        {
+            return -1;
+        }
+        if (dsc2->lvTracked)
+        {
+            return +1;
+        }
+
+        // Neither variable is tracked; separate GC from non-GC
+        if (varTypeIsGC(dsc1->TypeGet()))
+        {
+            if (!varTypeIsGC(dsc2->TypeGet()))
+            {
+                return -1;
+            }
+        }
+        else
+        {
+            if (varTypeIsGC(dsc2->TypeGet()))
+            {
+                return +1;
+            }
+        }
+
+        if (dsc1 < dsc2)
+        {
+            return -1;
+        }
+        if (dsc1 > dsc2)
+        {
+            return +1;
+        }
+
+        return 0;
+    }
+
+    // TODO(REVIEW): if we're here it means we have MinOpts and "simplified
+    // GC tracking", so we may be able to prune the stuff that follows.
```

Comment:
It seems like you should secondarily sort on weights, after GC-ness, before falling back to lclvar number to break ties.